subiquity/controllers/error.py

```diff
@@ -61,7 +61,7 @@ def new(cls, controller, kind):
             datetime.datetime.utcnow().isoformat(timespec='seconds'),
             kind.name.lower())
         i = 0
-        while 1:
+        while i < 10000:
```

Comment:
I guess I don't understand the failing scenarios.  Is subiquity waiting on something and it keeps trying until it's present?  If we're waiting, shouldn't we sleep in this loop?  If we're not sleeping in this loop will we hit 10k too fast?

Reply:
No, this is just the usual thing of atomically opening a file without overwriting an existing one. You do know what the 'x' flag means to open()?

Reply:
OK, but I don't understand the failure scenario.  Multiple crashes in the same subiquity run?  Surely using a time-based filename should be sufficient, something with micro-second resolution won't end up overlapping multiple calls.

Reply:
To some extent this is perhaps overthought as the expected number of crash files produced in any given run is at most 2 (of different kinds, to boot). But I do want the filenames to be human-friendly: they are saved to the install media by default, and if the user then pops their usb stick in their desktop to report a bug from there, I want them to be able to find them.

I guess I don't understand your concern here. The fact that this code is complex? That it has undesired consequences?

Reply:
BTW the most recent version just puts the date in the timestamp.